tests/libvmi/storage.go

```diff
@@ -0,0 +1,109 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package libvmi
+
+import (
+	kvirtv1 "kubevirt.io/client-go/api/v1"
+)
+
+// WithContainerImage specifies the name of the container image to be used.
+func WithContainerImage(name string) Option {
+	return func(vmi *kvirtv1.VirtualMachineInstance) {
+		diskName, bus := "disk0", "virtio"
+		vmi.Spec.Domain.Devices.Disks = append(vmi.Spec.Domain.Devices.Disks, newDisk(diskName, bus))
+		vmi.Spec.Volumes = append(vmi.Spec.Volumes, newContainerVolume(diskName, name))
+	}
+}
+
+func addDisk(vmi *kvirtv1.VirtualMachineInstance, disk kvirtv1.Disk) {
+	if !diskExists(vmi, disk) {
```

Comment:
@EdDev Any reason why to silently ignore options? Isn't it better to panic?

Reply:
@xpivarc , the `addDisk` (and probably also volume) are serving at the current implementation the could-init storage.
These are called from two functions, the one that handles the [user-data](https://github.com/kubevirt/kubevirt/blob/a9292cdb466d4e852299d809bc36b1edb350d35c/tests/libvmi/cloudinit.go#L29) and the one that uses the [network-data](https://github.com/kubevirt/kubevirt/blob/a9292cdb466d4e852299d809bc36b1edb350d35c/tests/libvmi/cloudinit.go#L45).

Therefore, the original idea was that if one option is added, the other should just re-use the resources that are already there.
Currently, the disks and volumes have been fixed per the task in hand. It also reflected the actual code that existed, perhaps with some exceptions.